src/URI/HierarchicalPart.purs

```diff
@@ -42,7 +42,9 @@ import URI.Path.Rootless as PathRootless
 -- | no ambiguity in parsing (this is per the spec, not a restriction of the
 -- | library).
 data HierarchicalPart userInfo hosts path hierPath
-  = HierarchicalPartAuth (Authority userInfo hosts) path
+  = HierarchicalPartAuth { authority :: Authority userInfo hosts
```

Comment:
This is still using `data` instead of `newtype`.

Reply:
That's because the full definition is a sum type:
```purescript
data HierarchicalPart userInfo hosts path hierPath
  = HierarchicalPartAuth { authority :: Authority userInfo hosts
                         , path :: path
                         }
  | HierarchicalPartNoAuth (Maybe hierPath)
```

Reply:
Oops!